### ISSUE TITLE
Uncaught Error: Call to a member function getValue() on boolean in

### ISSUE THREAD

**Issue by shuaiZend** [magento/magento2#3855]
## Steps to reproduce
1. Install Magento from `2.0` branch.   On **php7**
2. Add Related Products to the Product.
## Expected result
1. view Product page
## Actual result
1.  Error message appears:

```
<b>Fatal error</b>:  Uncaught Error: Call to a member function getValue() on boolean in /home/wwwroot/mysite/vendor/magento/module-catalog/Pricing/Render/FinalPriceBox.php:101
Stack trace:
#0 /home/wwwroot/mysite/vendor/magento/module-catalog/view/base/templates/product/price/final_price.phtml(24): Magento\Catalog\Pricing\Render\FinalPriceBox-&gt;hasSpecialPrice()
#1 /home/wwwroot/mysite/vendor/magento/framework/View/TemplateEngine/Php.php(59): include('/home/wwwroot/i...')
#2 /home/wwwroot/mysite/vendor/magento/framework/View/Element/Template.php(255): Magento\Framework\View\TemplateEngine\Php-&gt;render(Object(Magento\Catalog\Pricing\Render\FinalPriceBox), '/home/wwwroot/i...', Array)
#3 /home/wwwroot/mysite/vendor/magento/framework/View/Element/Template.php(279): Magento\Framework\View\Element\Template-&gt;fetchView('/home/wwwroot/i...')
#4 /home/wwwroot/mysite/vendor/magento/framework/Pricing/Render/PriceBox.php(65): Magento\Framework\View\Element\Template-&gt;_toHtml()
#5 /home/wwwroot/mysite/vendor/ma in <b>/home/wwwroot/mysite/vendor/magento/module-catalog/Pricing/Render/FinalPriceBox.php</b> on line <b>101</b><br />
```

**Comment by tomvolz** [comment]
same problem here. Do you have a solution for this?

**Comment by shuaiZend** [comment]
@tomvolz remove Special price.

**Comment by quynhvv** [comment]
@guys: How to solve this issue?
I get this issue too.
My context: PHP 5.6.20 + Nginx server + Magento 2.0.7
Thanks.

**Comment by SilvanLaroo** [comment]
I have this issue when showing the new products widget on my homepage via Magento_Catalog/templates/product/widget/new/content/new_grid.phtml. This only happens for configurable products.

**Comment by piotrekkaminski** [comment]
@SilvanLaroo i'm not able to reproduce this issue. can you check if still happens at 2.1rc2 or develop version?

**Comment by palamar** [comment]
According to [contributor guide](http://devdocs.magento.com/guides/v2.1/contributor-guide/contributing.html), tickets without response for two weeks should be closed.
Please feel free to reopen if it's needed.

**Comment by inkxe-jitendra** [comment]
@palamar i get this issue too.magento version is 2.0.7

Fatal error: Call to a member function getValue() on a non-object in /home/demoinkxe/public_html/magento2/vendor/magento/module-catalog/Pricing/Render/FinalPriceBox.php on line 101

**Comment by giacmir** [comment]
This is due to `getMinRegularAmount` returning `false` instead of an instance of `AmountInterface` in `ConfigurableRegularPrice` class. It seems to be fixed in 2.1.

For Magento 2.0.x I solved by adding an after plugin to that method that checks if the result is wrong and then fixes it by using the `AbstractPrice` implementation of that method.

**Comment by akshgods** [comment]
@giacmir , how you solved this issue. im facing on 2.0.4 . can you please give brief to solve it?
